torch/csrc/distributed/c10d/init.cpp

```diff
@@ -1477,21 +1477,21 @@ that adds a prefix to each key inserted to the store.
                 return ::c10d::ops::alltoall(
                     self, output_tensors, input_tensors, opts);
               },
-              py::arg("output_tensor"),
-              py::arg("input_tensor"),
+              py::arg("output_tensors"),
+              py::arg("input_tensors"),
               py::arg("opts") = ::c10d::AllToAllOptions(),
               py::call_guard<py::gil_scoped_release>())
 
           .def(
               "alltoall",
               [](const c10::intrusive_ptr<::c10d::ProcessGroup>& self,
-                 std::vector<at::Tensor>& output,
-                 std::vector<at::Tensor>& input) {
+                 const std::vector<at::Tensor>& output_tensors,
+                 const std::vector<at::Tensor>& input_tensors) {
```

Comment:
Do we expect calling this through `pg.alltoall(..)` to be backward compatible?

Reply:
good point, yeah that BC breaking issue is weird, let me investigate a bit what happened there

Reply:
The BC error is on the return type for `alltoall_`. Looks like this is recently added into BC check. This should be fine, as it is not a user-facing API, not even user-facing for PG extensions devs. 

```
2022-12-09T03:04:38.6291268Z processing existing schema:  c10d::alltoall_(Tensor[] _0, Tensor[] _1, __torch__.torch.classes.c10d.ProcessGroup _2, int _3) -> __torch__.torch.classes.c10d.Work _0
2022-12-09T03:04:38.6292120Z Can NOT find forward compatible schemas after changes for schema c10d::alltoall_(Tensor[] _0, Tensor[] _1, __torch__.torch.classes.c10d.ProcessGroup _2, int _3) -> __torch__.torch.classes.c10d.Work _0 from the following candidates:
2022-12-09T03:04:38.6292185Z [
2022-12-09T03:04:38.6292525Z c10d::alltoall_(Tensor[] _0, Tensor[] _1, __torch__.torch.classes.c10d.ProcessGroup _2, int _3) -> (Tensor[] _0, __torch__.torch.classes.c10d.Work _1)
2022-12-09T03:04:38.6292634Z ]
```